src/redux/ui/screens/data/actions/transactions/transaction-builder.ts

```diff
@@ -0,0 +1,115 @@
+    params: IContractCallParams,
+    currentTransactions: IBlockchainTransaction[]
+): Promise<IBlockchainTransaction<any>> => {
+    const state = getState();
+    const account = getSelectedAccount(state);
+    const chainId = getChainId(state, account.blockchain);
+
+    const blockchainInstance = getBlockchain(account.blockchain);
+
+    if (isWhitelistedMethod(account.blockchain, params.contractMethod)) {
+        const contractAddress = await getContract(chainId, params.contractType);
+        const tokenConfig = getTokenConfig(account.blockchain, params.tokenSymbol);
+        const client = blockchainInstance.getClient(chainId);
+
+        const nonce = await client.getNonce(account.address, account.publicKey);
+
+        const raw = JSON.stringify({
+            _tag: params.contractMethod, // 'SwapExactTokensForZIL',
+            params: params.rawData
+        });
+
+        // [
+        //     {
+        //         vname: 'token_address',
+        //         type: 'ByStr20',
+        //         value: param.fromToken
+        //     },
+        //     {
+        //         vname: 'min_zil_amount',
+        //         type: 'Uint128',
+        //         value: param.toTokenAmount
+        //     },
+        //     {
+        //         vname: 'recipient_address',
+        //         type: 'ByStr20',
+        //         value: param.toToken
+        //     },
+        //     {
+        //         vname: 'token_amount',
+        //         type: 'Uint128',
+        //         value: param.fromTokenAmount
+        //     },
+        //     {
+        //         vname: 'deadline_block',
+        //         type: 'Uint128',
+        //         value: param.blockNumber + 100 // expire after 100 blocks
+        //     }
+        // ]
+
+        const feeOptions = await client.getFees(
+            TransactionType.CONTRACT_CALL,
+            {
+                from: account.address,
+                to: '',
+                amount: params.amount,
+                contractAddress,
+                raw
+            },
+            tokenConfig.type
+        );
+
+        return {
+            date: {
+                created: Date.now(),
+                signed: Date.now(),
+                broadcasted: Date.now(),
+                confirmed: Date.now()
+            },
+            blockchain: account.blockchain,
+            chainId,
+            type: TransactionType.CONTRACT_CALL,
+            token: tokenConfig,
+            address: account.address,
+            publicKey: account.publicKey,
+            toAddress: contractAddress,
+            amount: params.amount,
+            feeOptions,
+            broadcastedOnBlock: params.blockNumber,
+            nonce: nonce + currentTransactions.length,
+            status: TransactionStatus.PENDING,
+            data: {
+                method: params.contractType + ' ' + tokenConfig.symbol, // TODO change to somehting relevant
+                params: [contractAddress, params.amount],
```

Comment:
this should contain the list of args that were used when calling the method

Reply:
we use this in transaction details, first index is the address where the call is made, second is the amount. We need to change how we handle the transaction details screen if we want to adjust this